cranelift-wasm/src/environ/spec.rs

```diff
@@ -141,6 +129,23 @@ pub trait FuncEnvironment {
             _ => panic!("unsupported pointer type"),
         }
     }
+}
+
+/// Environment affecting the translation of a single WebAssembly function.
+///
+/// A `FuncEnvironment` trait object is required to translate a WebAssembly function to Cranelift
+/// IR. The function environment provides information about the WebAssembly module as well as the
+/// runtime environment.
+pub trait FuncEnvironment: PointerEnvironment {
+    /// Get the information needed to produce Cranelift IR for the given target.
+    fn target_config(&self) -> TargetFrontendConfig;
```

Comment:
Since `PointerEnvironment` has this method, it shouldn't be necessary for `FuncEnvironment` to have it anymroe. I kind of wanted to factor this method out into its own trait anyways, since `ModuleEnvironment` requires it as well.